### PR TITLE
Issue #1566: HiddenField violations fixed

### PR DESCRIPTION
**Pull request by rdiachenko**
Violations fixed:
- HiddenField 'X' hides a field.

HiddenField gave us 350 violations. After configuring the check with the following options:
```
<property name="ignoreConstructorParameter" value="true"/>
<property name="ignoreSetter" value="true"/>
<property name="setterCanReturnItsClass" value="true"/>
```

the number of violations were reduced to 75 which were fixed manually.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java [hunk 1]
```diff
@@ -415,10 +415,10 @@ public final void setSeverity(String severity) {
      * Some Check implementations will use that classloader to improve the
      * quality of their reports, e.g. to load a class and then analyze it via
      * reflection.
-     * @param loader the new classloader
+     * @param classLoader the new classloader
      */
-    public final void setClassLoader(ClassLoader loader) {
-        this.loader = loader;
+    public final void setClassLoader(ClassLoader classLoader) {
+        this.loader = classLoader;
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java]: this is just a setter, why we need to update it?

**Comment by rdiachenko** [comment]
@mkordas 
>> this is just a setter, why we need to update it?

Setter name should be the same as the field name in order so the setter parameter could be skipped from validation. That is a good catch actually. I had to rename the field this setter operates with rather than the parameter name itself. I pushed the changes.

**Comment by mkordas** [comment]
@rdiachenko - there's one more thing:
```java
  public final void setClassLoader(ClassLoader loader) {
        this.classLoader = loader;
     }
```
by this one you introduce obsolete `this`. Let's also rename parameter name to `classLoader`.

**Comment by rdiachenko** [comment]
@mkordas done

**Comment by romani** [comment]
merged as FF

**Pull request closed without merging**